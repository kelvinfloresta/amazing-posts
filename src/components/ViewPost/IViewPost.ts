import Post from "entities/Post.entity";

export interface IViewPostLoadedProps {
  post: Post;
  isLoading?: boolean;
  onClick?: () => void;
}

export interface IViewPostLoadingProps {
  post?: null | Post;
  isLoading: true;
  onClick?: () => void;
}
