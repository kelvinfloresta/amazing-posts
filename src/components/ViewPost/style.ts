import styled from "styled-components";

export const PostTitle = styled.h4`
  font-size: 1.125rem;
  font-weight: 700;
  line-height: 1.688rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: ${(props) => props.theme.textColor1};
`;

export const PostDescription = styled.p`
  margin: 0;
  color: ${(props) => props.theme.textColor2};
`;
