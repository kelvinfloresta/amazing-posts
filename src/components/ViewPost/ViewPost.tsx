import React from "react";
import { IViewPostLoadingProps, IViewPostLoadedProps } from "./IViewPost";
import Card from "../Card/Card";
import { PostTitle, PostDescription } from "./style";
import useTextEllipsis from "containers/useTextEllipsis";
import { SkeletonTitle, SkeletonParagraph } from "components/Skeleton";

export function ViewPost(props: IViewPostLoadedProps | IViewPostLoadingProps) {
  const description = useTextEllipsis(props.post?.body);
  if (props.isLoading) {
    return (
      <Card height="200px">
        <SkeletonTitle />
        <SkeletonParagraph count={5} />
      </Card>
    );
  }
  const { post, onClick } = props;
  return (
    <Card height="200px" onClick={onClick}>
      <PostTitle title={post.title}>{post.title}</PostTitle>
      <PostDescription>{description}</PostDescription>
    </Card>
  );
}
