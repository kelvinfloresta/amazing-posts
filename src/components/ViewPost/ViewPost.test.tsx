import React from "react";
import renderer from "react-test-renderer";
import { ViewPost } from "./ViewPost";

describe("ViewPost", () => {
  it("should render loading correctly", () => {
    const tree = renderer.create(<ViewPost isLoading />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render post correctly", () => {
    const post = {
      id: 1,
      body: "Body",
      title: "Title",
    };
    const tree = renderer
      .create(<ViewPost post={post} onClick={() => console.log("clicked")} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render body with ellipsed text", () => {
    const post = {
      id: 1,
      body: new Array(10).fill("Really long text").join(", "),
      title: "Title",
    };
    const tree = renderer
      .create(<ViewPost post={post} onClick={() => console.log("clicked")} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
