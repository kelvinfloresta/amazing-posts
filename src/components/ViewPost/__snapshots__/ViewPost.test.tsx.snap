// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPost should render body with ellipsed text 1`] = `
<div
  className="sc-AxjAm clPXtE"
  height="200px"
  onClick={[Function]}
>
  <h4
    className="sc-AxirZ cntdGL"
    title="Title"
  >
    Title
  </h4>
  <p
    className="sc-AxiKw hQAPTQ"
  >
    Really long text, Really long text, Really long text, Really long text, Really long text, Really long text, Really long text, Really long text, Really...
  </p>
</div>
`;

exports[`ViewPost should render loading correctly 1`] = `
<div
  className="sc-AxjAm jUiYuM"
  height="200px"
>
  <span>
    <span
      className="react-loading-skeleton sc-AxhCb jBa-DGO css-1vmnjpn-skeletonStyles-Skeleton"
      style={Object {}}
    >
      ‌
    </span>
  </span>
  <span>
    <span
      className="react-loading-skeleton sc-AxhUy jQJchb css-1vmnjpn-skeletonStyles-Skeleton"
      style={Object {}}
    >
      ‌
    </span>
    <span
      className="react-loading-skeleton sc-AxhUy jQJchb css-1vmnjpn-skeletonStyles-Skeleton"
      style={Object {}}
    >
      ‌
    </span>
    <span
      className="react-loading-skeleton sc-AxhUy jQJchb css-1vmnjpn-skeletonStyles-Skeleton"
      style={Object {}}
    >
      ‌
    </span>
    <span
      className="react-loading-skeleton sc-AxhUy jQJchb css-1vmnjpn-skeletonStyles-Skeleton"
      style={Object {}}
    >
      ‌
    </span>
    <span
      className="react-loading-skeleton sc-AxhUy jQJchb css-1vmnjpn-skeletonStyles-Skeleton"
      style={Object {}}
    >
      ‌
    </span>
  </span>
</div>
`;

exports[`ViewPost should render post correctly 1`] = `
<div
  className="sc-AxjAm clPXtE"
  height="200px"
  onClick={[Function]}
>
  <h4
    className="sc-AxirZ cntdGL"
    title="Title"
  >
    Title
  </h4>
  <p
    className="sc-AxiKw hQAPTQ"
  >
    Body
  </p>
</div>
`;
